# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

235.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific  ...  compreq  orgreq  manuf  transcom  bankreal trade  \
0             17  ...      1.0     0.0    1.0       0.0       0.0   0.0   
1            316  ...      1.0     0.0    1.0       0.0       0.0   0.0   
2             19  ...      1.0     0.0    1.0       0.0       0.0   0.0   
3            313  ...      1.0     0.0    1.0       0.0       0.0   0.0   
4            313  ...      1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [7]:
data.shape

(4870, 65)

In [6]:
data.columns

Index(['id', 'ad', 'education', 'ofjobs', 'yearsexp', 'honors', 'volunteer',
       'military', 'empholes', 'occupspecific', 'occupbroad', 'workinschool',
       'email', 'computerskills', 'specialskills', 'firstname', 'sex', 'race',
       'h', 'l', 'call', 'city', 'kind', 'adid', 'fracblack', 'fracwhite',
       'lmedhhinc', 'fracdropout', 'fraccolp', 'linc', 'col', 'expminreq',
       'schoolreq', 'eoe', 'parent_sales', 'parent_emp', 'branch_sales',
       'branch_emp', 'fed', 'fracblack_empzip', 'fracwhite_empzip',
       'lmedhhinc_empzip', 'fracdropout_empzip', 'fraccolp_empzip',
       'linc_empzip', 'manager', 'supervisor', 'secretary', 'offsupport',
       'salesrep', 'retailsales', 'req', 'expreq', 'comreq', 'educreq',
       'compreq', 'orgreq', 'manuf', 'transcom', 'bankreal', 'trade',
       'busservice', 'othservice', 'missind', 'ownership'],
      dtype='object')

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

* What test is appropriate for this problem? Does CLT apply?
* What are the null and alternate hypotheses?

* Two-sample t-test is appropiate because we are trying to determine if two population means are equal.CLT applies because our sample size is large (n = 4870) and observations are independent.
* the null hypothesis is that there is no significant difference between the two sample means and the alternate hypothesis is there is a difference between the 2 sample means

In [5]:
w = data[data.race=='w']
b = data[data.race=='b']

*** Q3- Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.**

In [8]:
# Separate into two datasets
white = data[data['race']=='w']
black = data[data['race']=='b']

# Number of resumes
n_w = len(white)
n_b = len(black)

# Proportion of callbacks
prop_w = np.sum(white['call']) / n_w
prop_b = np.sum(black['call']) / n_b
print('Percentage of callbacks for whites: ', prop_w)
print('Percentage of callbacks for blacks: ', prop_b)

# Difference in proportion of callbacks
prop_diff = prop_w - prop_b
print('Difference in percentage of callbacks: ', prop_diff)

# T-score
t_stat, p = stats.ttest_ind(white['call'],black['call'],equal_var=False)
print('t-statistic: ', t_stat)
print('p-value: ', p)

# Standard error
s_error = np.sqrt(white['call'].var()/n_w + black['call'].var()/n_b)

# Margin of Error
m_error = 1.96 * s_error
print('Margin of error:', m_error)

# Confidence Interval
c_int = prop_diff + (np.array([-1, 1]) * m_error)
print('Confidence interval:', c_int)

# p-value
p_value = stats.norm.cdf(-t_stat) * 2
print('p-value:', p_value)

prop_w/prop_b

Percentage of callbacks for whites:  0.09650924024640657
Percentage of callbacks for blacks:  0.06447638603696099
Difference in percentage of callbacks:  0.032032854209445585
t-statistic:  4.114705290861751
p-value:  3.942941513645935e-05
Margin of error: 0.015258417380692
Confidence interval: [0.01677444 0.04729127]
p-value: 3.876744018936831e-05


1.4968152866242037

White-sounding names receive a callback rate of 0.0965, while black-sounding names receive only 0.0645. Could the difference we observe simply be a result of chance or does race affect callback percentage?

To answer this, we compute the $t$-statistic under the null hypothesis and find $t$ = 4.11, with p = 0.000039. This high $t$ score and very small p value lead us to reject the null hypothesis and conclude that white- and black-sounding names receive different callback rates.

The difference in callback rates is 0.0320. The margin of error on this difference is 0.0153. The 95% confidence interval on the difference in callback rates 0.0168 - 0.0473.

<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

#### Write a story describing the statistical significance in the context or the original problem.

 The results from the survey show that white-sounding names received a callback percentage of 9.65%, while black-sounding names received a callback percentage of only 6.45%.We wanted to know if this observed difference could be random. To answer this, we conducted a two sample $t$-test. Our analysis shows that the race of a candidate does have a statistically significant effect on his/her callback success.

### Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

Our analysis says that race is a factor in callback success; it does not say that race is the most important factor. In order to determine if race is the most important factor, we would need to account for all other variables (education, previous number of jobs, experience level, gaps in employment history, etc.)